src/Build/BackEnd/Client/MSBuildClient.cs

```diff
@@ -169,70 +169,86 @@ public MSBuildClientExitResult Execute(CancellationToken cancellationToken)
                 string.Join(" ", _commandLine);
 #endif
 
-            CommunicationsUtilities.Trace("Executing build with command line '{0}'", descriptiveCommandLine);
+            CommunicationsUtilities.Trace("Trying execute build at server with command line '{0}'", descriptiveCommandLine);
 
             try
             {
-                bool serverIsAlreadyRunning = ServerIsRunning();
-                if (KnownTelemetry.PartialBuildTelemetry != null)
+                ConfigureAndQueryConsoleProperties();
+
+                if (_consoleConfiguration?.OutputIsScreen == false && Environment.GetEnvironmentVariable(Traits.UseMSBuildServerInNonInteractiveEnvVarName) != "1")
```

Comment:
Understanding check:
You need a live env var check here because the server would cache it between builds, and we don't want that?